### PR TITLE
Jsap 3

### PR DESCRIPTION
**Pull request by eleven032**
transaction frontend page design done with store connect to database

### CODE REVIEW

File: frontend/src/containers/InternalTransactionScreenContainer.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { connect } from 'react-redux';
+import fetchAccountsAction from '../actions/fetchAccountsAction';
+import InternalTransactionScreen from '../screens/InternalTransactionScreen';
+
+const mapStateToProps = ({ userReducer, accountsReducer, transferAccountReducer }) => ({
+  token: userReducer.user.token,
```
Comment (matecserven) [inline frontend/src/containers/InternalTransactionScreenContainer.js]: why is this stored here as well?

**Comment by matecserven** [comment]
components/TransferScreen is enough dont need to write components again

File: frontend/src/components/TransferScreenComponents/DropDownAccountsList.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  const onValueChange = (value) => {
+    setSelect(value);
+    transferAccountSelection(value);
+  };
+
+  return (
+    <Card>
+      <CardItem>
+        <Text>To: </Text>
```
Comment (yuhao3570) [inline frontend/src/components/TransferScreenComponents/DropDownAccountsList.js]:
```suggestion
        <Text>To:fvfvf </Text>
```

File: frontend/src/components/TransferScreenComponents/ScanButton.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Text, Button, Icon } from 'native-base';
+import buttonStyle from '../../styles/BottomButtonStyle';
```
Comment (yuhao3570) [inline frontend/src/components/TransferScreenComponents/ScanButton.js]: BottomButtonStyle is used by multiple files, consider set individual style for buttons of different screen

File: frontend/src/components/TransferScreenComponents/DropDownAccountsList.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Text, Icon, Picker, Card, CardItem,
+} from 'native-base';
+
+export default function DropDownAccountsList({ accounts, transferAccountSelection }) {
+  const [selected, setSelect] = React.useState(undefined);
```
Comment (matecserven) [inline frontend/src/components/TransferScreenComponents/DropDownAccountsList.js]:
```suggestion
  const [selected, setSelected] = React.useState(undefined);
```
State name and setter should be same, also selected what? naming is not clear enough

File: frontend/src/screens/InternalTransactionScreen.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  const filterAccounts = accounts.filter((ele) => ele.depositName !== account.depositName);
+
+  return (
+    <Content padder>
+      <View>
+        <Card>
+          <CardItem>
+            <Text>
+              From:
+              {' '}
```
Comment (matecserven) [inline frontend/src/screens/InternalTransactionScreen.js]: is there no better way than this?

File: frontend/src/screens/QRCodeScannerScreen.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  handleBarCodeScanned = ({ type, data }) => {
+    this.setState({ scanned: true });
+    alert(`Bar code with type ${type} and data ${data} has been scanned!`);
+  };
+}
+
+const opacity = "rgba(0, 0, 0, .6)";
+const styles = StyleSheet.create({
```
Comment (matecserven) [inline frontend/src/screens/QRCodeScannerScreen.js]: put this in hte stylesheet file

File: frontend/src/screens/QRCodeScannerScreen.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import { Dimensions, Button, StyleSheet, Text, View } from 'react-native';
+import { Grid, Row } from "react-native-easy-grid";
+import * as Permissions from 'expo-permissions';
+import { BarCodeScanner } from 'expo-barcode-scanner';
+import { FontAwesome } from "@expo/vector-icons";
+export default class QRCodeScannerScreen extends React.Component {
```
Comment (matecserven) [inline frontend/src/screens/QRCodeScannerScreen.js]: convert this to functional component if its possible

File: frontend/src/screens/TransferScreen.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { View } from 'native-base';
+import TransferScreenLargeButton from '../components/TransferScreenComponents/TransferScreenLargeButton';
+import styles from '../styles/TransferScreenStyle';
+
+export default function TransferScreen() {
+  return (
+    <View style={styles.layout}>
+      <TransferScreenLargeButton screenName="InternalTransaction" context="BETWEEN YOUR ACCOUNTS" />
+      <TransferScreenLargeButton screenName="ExternalTransaction" context="TO OTHER ACCOUNTS" />
```
Comment (matecserven) [inline frontend/src/screens/TransferScreen.js]:
```suggestion
      <TransferScreenLargeButton screenName="ExternalTransaction" context="TO ANOTHER PERSON" />
```

**Pull request merged**